neurons/base_node.py

```diff
@@ -73,6 +83,106 @@ async def main(self):
     async def run(self):
         raise NotImplementedError
 
+    async def _run_diagnostics(self):
+        """Periodically collect and log diagnostic information"""
+        while not self.stop_event.is_set():
+            try:
+                await asyncio.sleep(300)  # 5 minutes
+
+                # Log summary
+                summary = []
+
+                # Memory
+                try:
+                    process = psutil.Process()
+                    mem_info = process.memory_info()
+                    summary.append(f"Memory: {mem_info.rss / (1024**3):.2f}GB")
+                except Exception:
+                    pass
+
+                # File descriptors
+                try:
+                    import resource
+
+                    soft_limit, _ = resource.getrlimit(resource.RLIMIT_NOFILE)
+                    try:
+                        open_fds = len(os.listdir("/proc/self/fd"))
+                        summary.append(f"FDs: {open_fds}/{soft_limit}")
+                    except Exception:
+                        pass
+                except Exception:
+                    pass
+
+                # S3 clients
+                if hasattr(self, "comms") and hasattr(self.comms, "_s3_clients"):
+                    summary.append(f"S3 Clients: {len(self.comms._s3_clients)}")
+
+                # Threads
+                summary.append(f"Threads: {threading.active_count()}")
+
+                # AsyncIO tasks
+                summary.append(f"Tasks: {len(asyncio.all_tasks())}")
+
+                # GPU
+                if torch.cuda.is_available():
```

Comment:
because we collect these already, why don't you just print them from the metrics? https://github.com/tplr-ai/templar/blob/2f6f072b77b21de08720446f9315ad0477fbc7e1/src/tplr/metrics.py#L333